subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/ToolChainRequirement.java

```diff
@@ -41,6 +41,12 @@
     GCC_COMPATIBLE,
     // Any available Clang
     CLANG,
-    // Any Swift Compiler
-    SWIFT
+    // Any compiler capable of compiling Objective-C source code
+    OBJECTIVE_C_SUPPORT,
+    // Any Swift compiler
+    SWIFT,
+    // Any Swift 3.x compiler
+    SWIFT3,
+    // Any Swift 4.x compiler
+    SWIFT4
```

Comment:
I feel like we should name those a bit better to reflect the difference between support for a specific language and a specific tool chain.